client/web/emberclear/app/components/pod/q-r/index.ts

```diff
@@ -1,80 +1,29 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
-import { inject as service } from '@ember/service';
 
-import { useMachine } from 'ember-statecharts';
-import { use } from 'ember-usable';
-
-import { machineConfig } from './-machine';
-import { LoginQRData } from './-types';
-
-import { SendDataConnection } from 'emberclear/services/connection/ephemeral/login/send-data';
-import CurrentUserService from 'emberclear/services/current-user';
-import { ConnectionDoesNotExistError } from 'emberclear/utils/errors';
-import QRManager from 'emberclear/services/qr-manager';
+import { StateMachine } from './state-machine';
 
 export default class QRScan extends Component {
-  @service intl!: Intl;
-  @service currentUser!: CurrentUserService;
-  @service qrManager!: QRManager;
-
-  @use interpreter = useMachine(machineConfig).withConfig({
-    services: {
-      setupConnection: this.setupConnection.bind(this),
-      transferData: this.transferData.bind(this),
-      addContact: this.addContact.bind(this),
-    },
-    guards: {
-      isQRLogin: ({ intent }) => intent === 'login',
-      isQRAddFriend: ({ intent }) => intent === 'add-friend',
-      hasError: ({ error }) => error === 'error',
-      isContactKnown: () => false,
-      isLoggedIn: () => this.currentUser.isLoggedIn,
-    },
-  });
+  // not implemented yet, but this should probably be an addon
+  @use machine = useDestroyable(StateMachine);
```

Comment:
@pzuraq, thoughts on a `useDestroyable`? 
would that be builtin? or should I proceed with an addon? :D 

I'm thinking it would just be in charge of
 - construction during `this`'s instantiation
 - destruction during `this`'s destruction
 - setOwner as well (for service injection)
    - this makes me think the name `useDestroyable` might not be a great fit